src/game/Person.java

```diff
@@ -41,8 +56,17 @@ public void receiveDamage(int damage)
     }
 
 
-
-    public int getRoomId() {return roomId; }
+    public void setRoomId(int roomId) {
+        if(roomId >= boardedShip.getRoomCount()) {
+            throw new RuntimeException("There is no room with this id");
```

Comment:
Lepiej `IllegalArgumentException`. W naszym przypadku nie ma to dużego znaczenia, ale ogólnie w Javie lepiej rzucać jak najbardziej dopasowane wyjątki. Często nawet na takie okazje robi się własne klasy wyjątków i się je rzuca, np. `NoRoomWithSpecifiedIdFoundException`, a takie wyjątki dziedziczą bezpośrednio po np. `IllegalArgumentException` albo nawet po samym `Exception`.

Reply:
Przykładowa implementacja takiego wyjątku:
`public class NoSuchRoomException extends Exception {
    public NoSuchRoomException(String message) {
        super(message);
    }
}`